src/allmydata/test/cli/test_invite.py

```diff
@@ -0,0 +1,320 @@
+            # about ignoring the "something-else" option
+            self.assertEqual(0, rc)
+            self.assertIn(u"option 'something-else' not whitelisted", out + err)
+
+
+class Invite(GridTestMixin, CLITestMixin, unittest.TestCase):
+
+    @defer.inlineCallbacks
+    def setUp(self):
+        self.basedir = self.mktemp()
+        yield super(Invite, self).setUp()
+        yield self.set_up_grid(oneshare=True)
+        intro_dir = os.path.join(self.basedir, "introducer")
+        yield run_cli(
+            "create-introducer",
+            "--listen", "none",
+            intro_dir,
+        )
+
+    @defer.inlineCallbacks
+    def test_invite_success(self):
+        intro_dir = os.path.join(self.basedir, "introducer")
+        # we've never run the introducer, so it hasn't created
+        # introducer.furl yet
+        priv_dir = join(intro_dir, "private")
+        with open(join(priv_dir, "introducer.furl"), "w") as f:
+            f.write("pb://fooblam\n")
+
+        with mock.patch('allmydata.scripts.tahoe_invite.wormhole') as w:
+            fake_wh = _create_fake_wormhole([
+                json.dumps({u"abilities": {u"client-v1": {}}}),
+            ])
+            w.create = mock.Mock(return_value=fake_wh)
+
+            rc, out, err = yield run_cli(
+                "-d", intro_dir,
+                "invite",
+                "--shares-needed", "1",
+                "--shares-happy", "1",
+                "--shares-total", "1",
+                "foo",
+            )
+            self.assertEqual(2, len(fake_wh.messages))
+            self.assertEqual(
+                json.loads(fake_wh.messages[0]),
+                {
+                    "abilities":
+                    {
+                        "server-v1": {}
+                    },
+                },
+            )
+            self.assertEqual(
+                json.loads(fake_wh.messages[1]),
+                {
+                    "shares-needed": "1",
+                    "shares-total": "1",
+                    "nickname": "foo",
+                    "introducer": "pb://fooblam",
+                    "shares-happy": "1",
+                },
+            )
+
+    @defer.inlineCallbacks
+    def test_invite_no_furl(self):
+        intro_dir = os.path.join(self.basedir, "introducer")
+
+        with mock.patch('allmydata.scripts.tahoe_invite.wormhole') as w:
+            fake_wh = _create_fake_wormhole([
+                json.dumps({u"abilities": {u"client-v1": {}}}),
+            ])
+            w.create = mock.Mock(return_value=fake_wh)
+
+            rc, out, err = yield run_cli(
+                "-d", intro_dir,
+                "invite",
+                "--shares-needed", "1",
+                "--shares-happy", "1",
+                "--shares-total", "1",
+                "foo",
+            )
+            self.assertNotEqual(rc, 0)
+            self.assertIn(u"Can't find introducer fURL", out + err)
```

Comment:
It seems like the `run_cli` args and following assertions are the only parts which vary between many of these tests.  Perhaps it would be nice to factor out the rest into a helper to reduce future maintenance burden and make the tests themselves easier to read.